examples/virtio/Makefile

```diff
@@ -155,44 +161,35 @@ $(BUILD_DIR)/package_guest_images.o: $(VMM_TOOLS)/package_guest_images.S $(IMAGE
 					-target aarch64-none-elf \
 					$< -o $@
 
-$(BUILD_DIR)/%.o: %.c Makefile
-	$(CC) -c $(CFLAGS) $< -o $@
-
-$(BUILD_DIR)/%.o: $(VMM_SRC_DIR)/%.c Makefile
-	$(CC) -c $(CFLAGS) $< -o $@
-
-$(BUILD_DIR)/%.o: $(VMM_SRC_DIR)/util/%.c Makefile
-	$(CC) -c $(CFLAGS) $< -o $@
+VPATH=${VMM_SRC_DIR}:${VMM_SRC_DIR}/util:${VMM_SRC_DIR}/arch/aarch64:${SDDF_SERIAL_DRIVER}:${SDDF_UTIL}:${VMM_SRC_DIR}/arch/aarch64/vgic:${VMM_SRC_DIR}/virtio
```

Comment:
What is this for? I could not find it used anywhere, is it a special Make variable or something?

Reply:
It's a standard Make thing to tell make where to look for source files.  Any precursor not in the current directory will be looked for in VPATH.  It allows removal of all the specific rules.

Reply:
What's more, unlike addprefix, it's in POSIX make....